ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -848,8 +955,366 @@ void adjust_conv_op_config_for_auto_shard_if_necessary(
         conv_config.act_block_w_div =
             tt::div_up(in_channels, width_sharded_num_cores * constants::TILE_WIDTH);
     }
+    return conv_config;
 }
 
+std::tuple<OptimizedConvParallelizationConfig, OptimizedConvBlockConfig, MemoryConfig> get_conv_configs(
```

Comment:
this should be a static method?

Reply:
`get_conv_configs` is also used in conv2d.cpp. Making it static would lead to an error.

Reply:
my bad I thought that this method is member funcion but it's a free function